Authorization: Web API protected by access token

rayluo opened the pull request:
Hi, @SHERMANOUKO, this PR draws inspiration from your previous one, incorporating most of its behaviors into this new implementation. Thank you very much for that. Regrettably, I failed to notify you earlier, but I had to start working on this PR over last weekend to address an urgent requirement for next week. I’ll share a sample PR shortly.

rayluo committed:
Authorization: Web API protected by access token

rayluo committed:
Merge ApiAuth into WebframeworkAuth for Django and Flask

rayluo committed:
Implement authorization_required() for Quart

rayluo committed:
Add docs for the Web API mode

omikader commented:
Hello @rayluo and @SHERMANOUKO. Do you know if this feature is still on the roadmap? I have a Flask web API that serves data to a React SPA that uses MSAL to authenticate. I would like to send a Bearer token in my API requests in order to validate and identify the requester using the token claims in the backend.

rayluo commented:
The feature is still in roadmap. It just takes time to finalize. This PR is considered as a preview.

rayluo committed:
Adding repo instructions for github copilot

https://docs.github.com/en/copilot/how-tos/configure-custom-instructions/add-repository-instructions#asking-copilot-coding-agent-to-generate-a-githubcopilot-instructionsmd-file
